File: sangjin/level1/13_I_don't_want_the_same_number.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    int preNum = -1;

    for(int i = 0; i < arr.size(); i++){
        if(preNum != arr[i]){
            answer.push_back(arr[i]);
        }
        preNum = arr[i];
    }

    return answer;
}

File: yunj/level1/13_I_don't_want_the_same_number.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int[] answer = {};
        
        ArrayList<Integer> temp = new ArrayList<>();
        temp.add(arr[0]);
        
        for(int a : arr){
            if (temp.get(temp.size()-1) != a)
                temp.add(a);
        }
        
        answer = new int[temp.size()];
        
        for(int i =0 ; i< answer.length; i++){
            answer[i] = temp.get(i);
        }
        
        return answer;
    }
}